### PR TITLE
expression: implement vectorized evaluation for builtinMicroSecondSig

### PR DESCRIPTION
**Pull request by YourLovely**
### What problem does this PR solve? 
expression: implement vectorized evaluation for `builtinMicroSecondSig` for #12103.

### What is changed and how it works?
```
goos: linux
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinTimeFuncGenerated-4   	1000000000	         0.00774 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinTimeFunc/builtinMicroSecondSig-VecBuiltinFunc-4         	  139454	      8298 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinTimeFunc/builtinMicroSecondSig-NonVecBuiltinFunc-4      	   43777	     27107 ns/op	       0 B/op	       0 allocs/op
PASS
ok  	github.com/pingcap/tidb/expression	2.785s
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by YourLovely** [comment]
@francis0407

File: expression/builtin_time_vec.go [hunk 1]
```diff
@@ -575,13 +575,33 @@ func (b *builtinFromDaysSig) vecEvalTime(input *chunk.Chunk, result *chunk.Colum
...
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		i64s[i] = int64(buf.GetDuration(i, int(types.UnspecifiedFsp)).MicroSecond())
+	}
+	return nil
 }
 
+
```
Comment (francis0407) [inline expression/builtin_time_vec.go]: Please remove this blank line.

Comment (YourLovely) [reply expression/builtin_time_vec.go]: ok

**Comment by YourLovely** [comment]
@francis0407

**Comment by francis0407** [comment]
The conflicts are not resolved yet. Run `git pull upstream master` to merge the master into your branch and resolve the conflicts.

**Comment by YourLovely** [comment]
> The conflicts are not resolved yet. Run `git pull upstream master` to merge the master into your branch and resolve the conflicts.

Ok.The conflicts are resolved.

File: expression/builtin_time_vec.go [hunk 2]
```diff
@@ -825,11 +825,30 @@ func (b *builtinFromDaysSig) vecEvalTime(input *chunk.Chunk, result *chunk.Colum
...
+	}
+
+	result.ResizeInt64(n, false)
+	result.MergeNulls(buf)
+	i64s := result.Int64s()
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		i64s[i] = int64(buf.GetDuration(i, int(types.UnspecifiedFsp)).MicroSecond())
```
Comment (francis0407) [inline expression/builtin_time_vec.go]: We don't need the `Fsp` here, you should use `buf.GoDurations()` to get the slice of Duration.

Comment (YourLovely) [reply expression/builtin_time_vec.go]: I have changed it according to your requirements, but I failed the test. What should I do？Can you help me?@francis0407

**Pull request merged**